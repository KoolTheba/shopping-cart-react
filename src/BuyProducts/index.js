import React from 'react'
import PropTypes from 'prop-types'

import { makeStyles } from '@material-ui/core/styles'
import Table from '@material-ui/core/Table'
import TableBody from '@material-ui/core/TableBody'
import TableContainer from '@material-ui/core/TableContainer'
import TableHead from '@material-ui/core/TableHead'
import TableRow from '@material-ui/core/TableRow'
import TableCell from '@material-ui/core/TableCell'

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faShoppingBasket } from '@fortawesome/free-solid-svg-icons'

import BuyProductItem from '../BuyProductItem'
import CleanCart from '../CleanCart'

const useStyles = makeStyles({
    buyProducts: {
      width: '100%',
      margin: '125px 0 50px',
    },
    table: {
      minWidth: 650,
      background: 'linear-gradient(45deg, #799F0C 30%, #799F0C 90%)',
      border: 0,
      borderRadius: 3,
      boxShadow: '0 3px 5px 2px rgba(255, 105, 135, .3)',
      color: 'white',
      height: 48,
      padding: '0 30px',
    },
    tableHead: {
        color: '#292C34',
        fontSize: 20,
        fontWeight: "bold",
        fontFamily: 'sans-serif',
    }
});

function BuyProducts (props) {
    const classes = useStyles()

    const productsList = Object.keys(props.cartList).map(id => {
        const {image, name, price} = props.products[id];
        return {
            id,
            quantity: props.cartList[id],
            image, 
            name, 
            price
        }
    })

    return(
        <>
        <section className={classes.buyProducts} id='products-list-section'>
            <h3>Your shopping bag <FontAwesomeIcon icon={faShoppingBasket}/></h3>
            <TableContainer>
                <Table className={classes.table} aria-label="simple table">
                    <TableHead>
                        <TableRow>
                            <TableCell className={classes.tableHead} align="center">Product</TableCell>
                            <TableCell className={classes.tableHead} align='left'></TableCell>
                            <TableCell className={classes.tableHead} align="center">Remove</TableCell>
                            <TableCell className={classes.tableHead} align="center">Unit price</TableCell>
                            <TableCell className={classes.tableHead} align="center">Quantity</TableCell>
                            <TableCell className={classes.tableHead} align="center">Total</TableCell>
                        </TableRow>
                    </TableHead>
                    <TableBody>
                        {productsList.length > 0 &&
                            productsList.map(product => (
                                <BuyProductItem
                                    key={product.id}
                                    product={product}
                                    addUnit={props.addUnit}
                                    removeUnit={props.removeUnit}
                                    removeListItem={props.removeListItem}
                                />
                            ))
                        }
                    </TableBody>
                </Table>
            </TableContainer>
            <CleanCart 
                cleanShoppingList={props.cleanShoppingList}
            />
        </section>
        </>
    )
}

BuyProducts.propTypes = {
    cartList: PropTypes.object,
    products: PropTypes.object,
    addUnit: PropTypes.func,
    removeUnit: PropTypes.func,
    removeListItem: PropTypes.func,
    cleanShoppingList: PropTypes.func,
}

export default BuyProducts