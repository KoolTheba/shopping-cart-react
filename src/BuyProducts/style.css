.products-to-buy {
    width: 100%;
    border: 1px solid white;
}

.table-head {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}