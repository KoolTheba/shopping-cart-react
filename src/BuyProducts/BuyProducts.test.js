import React from 'react'
import BuyProducts from './'
import { shallow, mount } from 'enzyme'

const props = {
    products: {
        "PLANT14TGG": {
            "name": "Zamioculcas",
            "price": 90.00,
            "stock": true,
            "stockQ": 15,
            "image": "https://image.shutterstock.com/image-photo/pot-zamioculcas-home-plant-on-600w-1399949501.jpg"
        },
    },
    cartList: {
        "PLANT14TGG": 2
    },
    removeListItem: jest.fn(),
    removeUnit: jest.fn(),
    addUnit: jest.fn(),
    cleanShoppingList: jest.fn()
}

describe('BuyProducts component', () => {
    it('renders BuyProducts component correctly', () => {
        const wrapper = shallow(<BuyProducts {...props}/>)
        expect(wrapper.find('#products-list-section')).toHaveLength(1)
        expect(wrapper).toMatchSnapshot()
    });

    it('renders no-products-info paragraph correctly when there are no products in the list', () => {
        const currentProps = {
            products: {},
            cartList: {}
        }
        const wrapper = shallow(<BuyProducts {...currentProps}/>)
        expect(wrapper.find('.no-products-info')).toHaveLength(1)
        expect(wrapper.find('.no-products-info').text()).toEqual('There are no products in your bag')
    });

    it('renders BuyProductsItem component correctly when there is at least 1 product and has props', () => {
        const productDetail = {
            id: 'PLANT14TGG',
            quantity: props.cartList['PLANT14TGG'],
            image: props.products['PLANT14TGG'].image,
            name: props.products['PLANT14TGG'].name,
            price: props.products['PLANT14TGG'].price
        }
        const wrapper = mount(<BuyProducts {...props}/>)
        expect(wrapper.find('BuyProductItem')).toHaveLength(1)
        expect(wrapper.find('BuyProductItem').props().product).toEqual(productDetail)
        expect(wrapper.find('BuyProductItem').props().addUnit).toEqual(props.addUnit)
        expect(wrapper.find('BuyProductItem').props().removeUnit).toEqual(props.removeUnit)
        expect(wrapper.find('BuyProductItem').props().removeListItem).toEqual(props.removeListItem)
    });

    it('renders CleanCart component correctly and has props...', () => {
        const wrapper = mount(<BuyProducts {...props}/>)
        expect(wrapper.find('CleanCart')).toHaveLength(1)
        expect(wrapper.find('CleanCart').props().cleanShoppingList).toEqual(props.cleanShoppingList)
    });
});

