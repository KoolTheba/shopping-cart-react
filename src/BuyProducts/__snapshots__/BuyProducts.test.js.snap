// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuyProducts component renders BuyProducts component correctly 1`] = `ShallowWrapper {}`;
